source/gui/guiHelper.py

```diff
@@ -124,12 +124,30 @@ def addButton(self, *args, **kwargs):
 		return wxButton
 
 
-def associateElements(firstElement: wx.Control, secondElement: wx.Control) -> wx.BoxSizer:
+# vertical controls where the label should go above visually, and the control should go below
+_VerticalCtrlT = TypeVar("_VerticalCtrlT", wx.ListCtrl, wx.ListBox, wx.TreeCtrl)
+# horizontal controls where the label should go first visually, and the control should go after
+_HorizontalCtrlT = TypeVar(
+	"_HorizontalCtrlT",
+	wx.Button,
+	wx.Choice,
+	wx.ComboBox,
+	wx.Slider,
+	wx.SpinCtrl,
+	wx.TextCtrl,
+)
+
+
+def associateElements(
+	firstElement: wx.StaticText | wx.StaticText | wx.StaticText | wx.Button | wx.TextCtrl,
```

Comment:
Why is `wx.StaticText` repeated so many times?
```suggestion
	firstElement: wx.StaticText | wx.Button | wx.TextCtrl,
```

Reply:
this is so its clear that these sequential types for each element match. The first type of each param matches, same with second, etc. I couldn't find another way to explicitly type pairs of argument types

Reply:
Is such mapping recognized by your dev environment?

If not, I think there is no point in listing precisely the possible associations in type hints. When reading the type hints, it's not obvious that there is a 1-to-1 mapping between the 2 arguments. And when I finally realize that there is a mapping, if I am looking at the n-th item in the first sequence and want to know what is the corresponding item in the second sequence, I need to manually count the items in the sequences; it's quite tedious.
The conclusion is that writing type hints this way just produces more verbose code without bringing any real benefit.

Reply:
I have fixed this using `overload`